apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php

```diff
@@ -131,6 +131,7 @@ function wpcom_block_global_styles_frontend( $theme_json ) {
 	return $theme_json;
 }
 add_filter( 'theme_json_user', 'wpcom_block_global_styles_frontend' );
+add_filter( 'wp_theme_json_data_user', 'wpcom_block_global_styles_frontend' );
```

Comment:
Is there a reliable way of registering only one of the filters? Would it be possible to conditionally register the new hook based on Gutenberg's version instead of registering both?

Reply:
The end result would be the same, because if the filter doesn't exist it'll never run so we don't need to worry about registering it only when it exists.

Reply:
Is there a chance of both filters existing (e.g. for fallback reasons) and we running the code twice for no reason?

Reply:
No. `theme_json_user` only exists in Gutenberg versions between 14.1.0 and 14.3.1 and Core versions between 6.1 beta 1 and 6.1 beta 3.

In order to have both filters, a site should be running WordPress 6.1 RC2 for instance (which has the new filter) and Gutenberg 14.2.0 (which has the previous filter).

However, only the one from Gutenberg will run because Gutenberg always overrides Core's default behavior: https://github.com/WordPress/gutenberg/blob/9493eee8a8c1cf4d06943d56109997861b1e715c/lib/compat/wordpress-6.1/script-loader.php#L174-L183

Reply:
Cool, thanks!